src/test/java/integration/EnrollmentTests.java

```diff
@@ -0,0 +1,153 @@
+import org.example.learningsystem.dto.student.StudentRequestDto;
+import org.example.learningsystem.service.course.CourseService;
+import org.example.learningsystem.service.student.StudentService;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@SpringBootTest(classes = {LearningSystemApplication.class})
+@AutoConfigureMockMvc
+@ActiveProfiles("test")
+public class EnrollmentTests {
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private CourseService courseService;
+    @Autowired
+    private StudentService studentService;
+    @Autowired
+    private ObjectMapper objectMapper;
+    private static Course course;
+    private static Student student;
+
+    @BeforeAll
+    static void setUp() {
+        course = new Course("Test course", "Course description", BigDecimal.valueOf(100));
+        course.setSettings(CourseSettings.builder().isPublic(true).build());
+
+        student = new Student("Name", "Surname", "name@gmail.com",
+                LocalDate.of(1900, 1, 1), BigDecimal.valueOf(150));
+    }
+
+    @Test
+    void enrollStudentSuccess() throws Exception {
+        /* Create course and student, make sure that the course is public,
+           and the number of student coins is not lower than the price
+         */
+        course.getSettings().setIsPublic(true);
+        course = createCourse(course);
+        student.setCoins(course.getPrice());
+        student = createStudent(student);
+
+        // Perform enrollment
+        mockMvc.perform(post("/courses/{id}/students/{studentId}", course.getId(), student.getId()))
+                .andExpect(status().isOk());
+
+        // Verify that course's students contain that student
+        var courseStudents = studentService.getAllByCourseId(course.getId());
+        assertEquals(1, courseStudents.size());
+        var updatedStudent = courseStudents.getFirst();
+        assertEquals(student.getId(),updatedStudent.getId());
+
+        // Verify that student's and course's value of coins changed
+        assertNotEquals(student.getCoins(), updatedStudent.getCoins());
+        var updatedCourse = courseService.getById(course.getId());
+        assertNotEquals(course.getCoinsPaid(), updatedCourse.getCoinsPaid());
```

Comment:
usually, comments are not written in tests
and it's better to split test method into 3 parts (and comment them):
```
// given
Student student = new Student();

// when
mockMvc.perform();

// then
assertEquals(1, ...);
```